cmd/src/debug_comp.go

```diff
@@ -0,0 +1,270 @@
+package main
+
+import (
+	"archive/zip"
+	"context"
+	"flag"
+	"fmt"
+	"log"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"sync"
+
+	"golang.org/x/sync/semaphore"
+
+	"github.com/sourcegraph/src-cli/internal/cmderrors"
+)
+
+func init() {
+	usage := `
+'src debug comp' mocks docker cli commands to gather information about a docker-compose Sourcegraph instance.
+
+Usage:
+
+    src debug comp [command options]
+
+Flags:
+
+	-o			Specify the name of the output zip archive.
+	-cfg		Include Sourcegraph configuration json. Defaults to true.
+
+Examples:
+
+    $ src debug comp -o debug.zip
+
+	$ src -v debug comp -cfg=false -o foo.zip
+
+`
+
+	flagSet := flag.NewFlagSet("comp", flag.ExitOnError)
+	var base string
+	var configs bool
+	flagSet.BoolVar(&configs, "cfg", true, "If true include Sourcegraph configuration files. Default value true.")
+	flagSet.StringVar(&base, "o", "debug.zip", "The name of the output zip archive")
+
+	handler := func(args []string) error {
+		if err := flagSet.Parse(args); err != nil {
+			return err
+		}
+
+		//validate out flag
+		if base == "" {
+			return fmt.Errorf("empty -o flag")
+		}
+		// declare basedir for archive file structure
+		var baseDir string
+		if !strings.HasSuffix(base, ".zip") {
+			baseDir = base
+			base = base + ".zip"
+		} else {
+			baseDir = strings.TrimSuffix(base, ".zip")
+		}
+
+		ctx := context.Background()
+
+		//Gather data for safety check
+		containers, err := getContainers(ctx)
+		if err != nil {
+			fmt.Errorf("failed to get containers for subcommand with err: %v", err)
+		}
+		// Safety check user knows what they are targeting with this debug command
+		log.Printf("This command will archive docker-cli data for %d containers\n SRC_ENDPOINT: %v\n Output filename: %v", len(containers), cfg.Endpoint, base)
+		if verified, _ := verify("Do you want to start writing to an archive?"); !verified {
+			return nil
+		}
+
+		out, zw, ctx, err := setupDebug(base)
```

Comment:
Note: here you're overwriting ctx, but you've already created a context on L64 above (this is related to my comment about inlining the setupDebug function [here](https://github.com/sourcegraph/src-cli/pull/731/files#r862014014)).

Reply:
refactored ✅